src/lib/replication/repl_dev/raft_repl_dev.cpp

```diff
@@ -953,18 +953,20 @@ void RaftReplDev::handle_commit(repl_req_ptr_t rreq, bool recovery) {
 
 void RaftReplDev::handle_error(repl_req_ptr_t const& rreq, ReplServiceError err) {
     if (err == ReplServiceError::OK) { return; }
+    RD_LOGE("Raft Channel: Error in processing rreq=[{}] error={}", rreq->to_string(), err);
 
     if (!rreq->add_state_if_not_already(repl_req_state_t::ERRORED)) {
-        RD_LOGE("Raft Channel: Error in processing rreq=[{}] error={}", rreq->to_string(), err);
+        RD_LOGE("Raft Channel: Error has been added for rreq=[{}] error={}", rreq->to_string(), err);
         return;
     }
 
     // Remove from the map and thus its no longer accessible from applier_create_req
     m_repl_key_req_map.erase(rreq->rkey());
+    // Remove from the lsn map and thus its no longger accessible for commit (is it really needed?)
+    m_state_machine->unlink_lsn_to_req(rreq->lsn(), rreq);
```

Comment:
the lsn of  a rreq is set after the corresponding log is appended.  and handle_error will always happen before log is appended. so , IMO, it will not find the lsn to the rreq and will do nothing. we can add an assert here to check if it has a lsn when handl_error of a rreq